File: client-app/src/interfaces/Recipe.ts
import { IUser } from './Common';

export interface IRecipeCard {
  id: number;
  title: string;
  imageUrl: string;
  preparationMinutes: number;
  cookingMinutes: number;
  likeCount: number | undefined;
  commentCount: number | undefined;
}

export interface IRecipeAdd {
  title: string;
  preparationMinutes: number;
  cookingMinutes: number;
  categoryId: number;
  description: string;
  ingredients: string[];
  instructions: string;
  imageFileName: string | null;
  isPublished: boolean;
}

export interface IRecipeEdit {
  id: number;
  title: string;
  preparationMinutes: number;
  cookingMinutes: number;
  categoryId: number;
  description: string;
  ingredients: string[];
  instructions: string;
  imageFileName: string | null;
  isPublished: boolean;
}

export interface IRecipe {
  id: number;
  categoryId: number;
  categoryName: string;
  title: string;
  description: string;
  imageUrl: string;
  imageFileName: string;
  preparationMinutes: number;
  cookingMinutes: number;
  ingredients: string[];
  instructions: string;
  user: IUser;
  createdAt: Date;
  updatedAt: Date;
  publishedAt: Date;
  isPublished: boolean;
}

File: Management/Management.Core/Entities/Recipe.cs
using System.Collections.ObjectModel;

namespace Management.Core.Entities;

public class Recipe
{
    public int Id { get; set; }

    public int CategoryId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string? ImageFileName { get; set; }

    public int PreparationMinutes { get; set; }

    public int CookingMinutes { get; set; }

    public Collection<string> Ingredients { get; set; } = [];

    public string Instructions { get; set; } = string.Empty;

    public string UserId { get; set; } = string.Empty;

    public bool IsPublished { get; set; }

    public bool IsDeleted { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public DateTime? PublishedAt { get; set; }
}
